
# Preliminary notes

The arguments telling Gviz to visually link exons into transcripts are documented but not necessarily easy to find.

# Visualize the exon structure of selected transcripts

`Gviz::GeneRegionTrack` expects information in columns that may be named differently in the GTF file:

- exon identifiers in `"exon"` (typically stored under `"exon_number"`)
- transcript identifiers in `"transcript"` (typically stored under `"transcript_id"`)
- gene identifiers in `"gene"` (typically stored under `"gene_id"`)

```{r}
# gtf : a GTF file
# which   : a `GRanges`
function(gtf, i) {
    gtf_exons <- import.gff(gtf_file, feature.type = "exon")
    # Prepare columns expected by Gviz
    gtf_exons$exon <- gtf_exons$exon_number
    gtf_exons$transcript <- gtf_exons$transcript_id
    gtf_exons$gene <- gtf_exons$gene_id
    # Split to visualize selected transcripts as examples
    gr_data <- split(gtf_exons, gtf_exons$transcript_id)
    gr_data <- gr_data[i]
    # Unlist to avoid an issue with GRangesList
    gr_data <- unlist(gr_data)
    # Make the track
    gr_track <-  GeneRegionTrack(
        range = gr_data,
        genome=gen, chromosome=chr,
        name="Track"
    )
    chr <- as.character(unique(seqnames(gr_track)))
    plotTracks(trackList = list(gr_track), chromosome = chr)
}
```
