
# Links

- https://support.bioconductor.org/p/86844/

# The `feature` argument

In the chunk below:

- `feature` takes a vector of length matching the number of elements in `range`
- `final` and `source` are values of `as.vector(example_grange$source)`
   which are mapped to the `"red"` and `"blue"` colors, respectively

```{r}
gen <- as.character(unique(genome(example_grange)))
chr <- as.character(unique(seqnames(example_grange)))
grtrack_combined <- GeneRegionTrack(
    range = example_grange,
    genome=gen, chromosome=chr, name="Genes",
    feature=as.vector(example_grange$source), final="red", source="blue"
)
plotTracks(trackList = list(gtrack, grtrack_combined))
```

